crates/ra_lsp_server/src/main_loop/handlers.rs

```diff
@@ -666,6 +662,50 @@ pub fn handle_code_action(
     Ok(Some(CodeActionResponse::Commands(res)))
 }
 
+pub fn handle_code_lens(
+    world: ServerWorld,
+    params: req::CodeLensParams,
+) -> Result<Option<Vec<CodeLens>>> {
+    let file_id = params.text_document.try_conv_with(&world)?;
+    let line_index = world.analysis().file_line_index(file_id);
+
+    let mut lenses: Vec<CodeLens> = Default::default();
+
+    for runnable in world.analysis().runnables(file_id)? {
+        match &runnable.kind {
+            RunnableKind::Test { name: _ } | RunnableKind::TestMod { path: _ } => {
+                let args = runnable_args(&world, file_id, &runnable.kind)?;
+
+                let range = runnable.range.conv_with(&line_index);
```

Comment:
I think this will cover the whole test funciton/module, but LSP spec says it must cover only one line. If this works, I am fine with it, but perhaps we should add a `indentifier_range` field to the `Runnable` and use it here.